### PR TITLE
Add 418 status code

### PR DESCRIPTION
**Pull request by pedantix**
Add "I'm teapot" status code to `HTTPResponseStatus`. And moved `unprocessableEntity` from 5XX section to 4XX section

### Motivation:

For the sake of pure transparency, this is hilarious, but not the sole reason for this PR. Two users other than myself of the Vapor project which is built atop Apple's excellent Swift NIO implementation are using custom response codes for the 418 status. Our current use case is a unique code that is used to check if the server is up.

### Modifications:

1. Moved `unprocessableEntity`(which is 422) from 500 section to 400 section.
2. Added `.imaTeapot` to the `HTTPResponseStatus` as the symbolic representation of "I'm a teapot", also added a case to the equality switch and enum to status code statement.

### Result:

1. This modification was just a passing item that stuck out like a sore thumb to me. This should have no effect on the code.
2. This modification will add `.iamTeapot` as a first class response status to `HTTPResponseStatus` instead of requiring a custom implementation of  418 in multiple projects.

### CODE REVIEW

**Comment by Lukasa** [comment]
@swift-nio-bot test this please

**Comment by normanmaurer** [comment]
@Lukasa that said I think this is a next-major as someone may have used switch on all cases (without default) right ?

**Comment by Lukasa** [comment]
Agreed.

**Comment by pedantix** [comment]
@Lukasa how do you feel about refactor opprotunities.  I see a major one in https://github.com/apple/swift-nio/blob/master/Sources/NIOHTTP1/HTTPTypes.swift
`HTTPResponseStatus: Equatable` is fairly ripe for refactoring I am pretty confident that  the following will have the same efficacy as the current implementation with a reduction of 100+ lines of code without loss of readability

```
extension HTTPResponseStatus: Equatable {
    public static func ==(lhs: HTTPResponseStatus, rhs: HTTPResponseStatus) -> Bool {
        switch (lhs, rhs) {
        case (.custom(let lcode, let lreason), .custom(let rcode, let rreason)):
            return lcode == rcode && lreason == rreason
        default:
            return lhs.code == rhs.code
        }
    }
}
```
 Should I fold it into here? Should I open a seperate PR or should I ignore it?

**Comment by normanmaurer** [comment]
@pedantix new PR please

**Comment by tanner0101** [comment]
Shouldn't this be `.imATeapot`?

**Comment by helje5** [comment]
so custom(418) is equal to .teapot, but custom(418,”earl gray”) differs to custom(418,”whatever”) 🤔

**Comment by pedantix** [comment]
@normanmaurer will do.

**Comment by pedantix** [comment]
@tanner0101 When I put this change together I came up with many possibilities, `.imaTeapot` seemed the most natural to me other possibilities were `.iMaTeapot`, `.imATeapot`, or `.iMATeapot` my enum foo is a bit weak, I wasn't sure which way it should go. If we remove conjunctions we get `.iAmATeapot`. I apologize if I selected wrong if the group things I should change I will.

File: Sources/NIOHTTP1/HTTPTypes.swift [hunk 1]
```diff
@@ -1371,10 +1375,11 @@ public enum HTTPResponseStatus {
     case unsupportedMediaType
     case rangeNotSatisfiable
     case expectationFailed
+    case imaTeapot
+    case unprocessableEntity
 
     // 5xx
     case misdirectedRequest
-    case unprocessableEntity
```
Comment (weissi) [inline Sources/NIOHTTP1/HTTPTypes.swift]: why is this removed?

Comment (Lukasa) [reply Sources/NIOHTTP1/HTTPTypes.swift]: It’s not, it’s moved to the 4xx section, which is where it belongs. (It’s 422.)

**Comment by weissi** [comment]
@Lukasa / @tanner0101 / @pedantix Happy with this, but how should we spell I’m a tea pot ;)

**Comment by helje5** [comment]
Swift is Unicode, so why not 我是一个茶壶 or imA🍵, fun code deserves fun representation.

**Comment by weissi** [comment]
@Lukasa shall we take this one?

**Comment by weissi** [comment]
@swift-nio-bot test this please

**Comment by weissi** [comment]
@tanner0101 want to make a PR? Could also spell it `iAmATeapot`

**Comment by tanner0101** [comment]
@weissi done here: https://github.com/apple/swift-nio/pull/860. Since the status phrase is specifically `"I'm a teapot"`, I think the correct spelling is `imATeapot`.
